src/core/api.proto

```diff
@@ -94,6 +94,55 @@ message InferRequestHeader
     //@@       for tensors with a non-fixed-size datatype (like STRING).
     //@@
     uint64 batch_byte_size = 3;
+
+    //@@      .. cpp:var:: message SharedMemoryElement
+    //@@
+    //@@         The Meta-data for the Input shared memory to read element from.
+    //@@
+    message SharedMemoryElement
+    {
+      //@@    .. cpp:var:: key_t shm_key
+      //@@
+      //@@     The unique key used to refer to a specific location in shared
```

Comment:
It's the name of the shared memory region that holds the input data (or where the output data should be written). It is not a specific location. You should drop the optional part. Specifying shared memory at all is optional, but if you add this message you must fill in all fields.